plugins/processors/t128_transform/t128_transform.go

```diff
@@ -105,11 +120,31 @@ func (r *T128Rate) Apply(in ...telegraf.Metric) []telegraf.Metric {
 	return in
 }
 
-func (r *T128Rate) Init() error {
+func (r *T128Transform) Init() error {
 	if len(r.Fields) == 0 {
 		return fmt.Errorf("at least one value must be specified in the 'fields' list")
 	}
 
+	switch r.Transform {
+	case "diff":
```

Comment:
Can you convert this to enumeration to avoid taking the hit of a string compare for each iteration. By converting to enumeration/integer you'll be taking hit only at config time as opposed to at run time.

Reply:
This IS only taking a hit at config time (note it's in the `Init` function). The intended operation is being determined here during start up and the appropriate function (essentially a lambda) is being stored into the `T128Transform.transform` variable. In the main code, that function is called with no conditional logic.